File: gpt-3.5-finetuned/stage2/kernels_construct_default_clause.f90
!T1:kernels construct default clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        
        INTEGER :: i, errors = 0
        REAL(8),DIMENSION(LOOPCOUNT):: a, b, c
        INTEGER :: d
        
        !$acc kernels loop default(none)
        DO i = 1, LOOPCOUNT
          c(i) = a(i) + b(i) * i
        END DO

        DO i = 1, LOOPCOUNT
          IF (abs(c(i) - (a(i) + b(i) * i)) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM

File: gpt-3.5-finetuned/stage2/kernels_construct_default_clause.c
#include "acc_testsuite.h"
#ifndef T1
//T1:kernels construct default clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    real_t * a = (real_t *)malloc(1024 * sizeof(real_t));
    real_t * b = (real_t *)malloc(1024 * sizeof(real_t));
    real_t * a_host = (real_t *)malloc(1024 * sizeof(real_t));
    real_t * b_host = (real_t *)malloc(1024 * sizeof(real_t));

    for (int x = 0; x < 1024; ++x){
        a[x] = (1024 * rand() / (real_t)(RAND_MAX));
        b[x] = 0.0;
        a_host[x] = a[x];
	b_host[x] = 0.0;
    }

    #pragma acc data copyin(a[0:1024]) copyout(b[0:1024])
    {
        #pragma acc kernels default(present)
	{
            #pragma acc loop
            for (int x = 0; x < 1024; ++x){
                b[x] = a[x];
            }
	}
    }

    for (int x = 0; x < 1024; ++x){
        if (fabs(a[x] - b[x]) > PRECISION){
            err += 1;
        }
    }

    for (int x = 0; x < 1024; ++x){
        if (fabs(a_host[x] - b_host[x]) > PRECISION){
            err += 1;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}